pennylane/devices/tf_ops.py

```diff
@@ -191,3 +191,56 @@ def CRot(a, b, c):
         :math:`|0\rangle\langle 0|\otimes \mathbb{I}+|1\rangle\langle 1|\otimes R(a,b,c)`
     """
     return tf.linalg.diag(CRZ(c)) @ (CRY(b) @ tf.linalg.diag(CRZ(a)))
+
+
+def SingleExcitation(phi):
+    r"""
+    Single excitation rotation.
+
+    Args:
+        phi (float): rotation angle
+
+    Returns:
+        tf.Tensor[complex]: Single excitation rotation matrix
+
+    """
+    phi = tf.cast(phi, dtype=C_DTYPE)
+    c = tf.cos(phi / 2)
+    s = tf.sin(phi / 2)
+    return tf.convert_to_tensor([[1, 0, 0, 0], [0, c, -s, 0], [0, s, c, 0], [0, 0, 0, 1]])

```

Comment:
Note: while true for JAX and autograd above, in this case the tensor _will_ be complex, since `phi` is cast to complex on line 208. So it should remain `tf.Tensor[complex]`.

Reply:
> Note: while true for JAX and autograd above, in this case the tensor _will_ be complex, since `phi` is cast to complex on line 208. So it should remain `tf.Tensor[complex]`.

Attention to details! :smile: